compiler/rustc_data_structures/src/sync/lock.rs

```diff
@@ -49,6 +49,18 @@ impl<T> Lock<T> {
         self.0.try_borrow_mut().ok()
     }
 
+    #[inline(always)]
+    #[track_caller]
+    pub unsafe fn lock_assume_no_sync(&self) -> LockGuard<'_, T> {
```

Comment:
This is safe when `#[cfg(not(parallel_compiler))]`. I think at least we need a comment here to explain this is to consist with same functions for `#[cfg(parallel_compiler)]`

Reply:
Done.